src/CommunityToolkit.Maui.Core/Essentials/Badge/Android/DefaultBadgeProvider.android.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Diagnostics;
+using Android.Content;
+using Android.Content.PM;
+
+namespace CommunityToolkit.Maui.ApplicationModel;
+
+class DefaultBadgeProvider : IBadgeProvider
+{
+	const string intentAction = "android.intent.action.BADGE_COUNT_UPDATE";
+
+	public void SetCount(int count)
+	{
+		if (count < 0)
+		{
+			return;
+		}
+
+		var packageName = Application.Context.PackageName;
+		if (packageName is null)
+		{
+			return;
+		}
+
+		var component = Application.Context.PackageManager?.GetLaunchIntentForPackage(packageName)?.Component;
+		if (component is null)
+		{
```

Comment:
Let's add a `Debug.WriteLine` here to let the dev know what happened instead of failing silently